pkg/tasks/probes.go

```diff
@@ -90,7 +90,9 @@ func safeguard(s *state.State) error {
 
 		if nodesContainerRuntime != configuredClusterContainerRuntime {
 			errMsg := "Migration is not supported yet"
-			if cr.Containerd != nil {
+			if nodesContainerRuntime == "docker" {
+				errMsg = "Support for docker will be removed with Kubernetes 1.24 release. It is recommended to switch to containerd as container runtime using `kubeone migrate to-containerd`. To continue using Docker please specify ContainerRuntime explicitly in KubeOneCluster manifest"
```

Comment:
Let me know what you think of this @xmudrii @kron4eg, I'm divided on this one. This is the scenario where a user running docker is trying to upgrade to k8s v.1.22.x. If he hasn't explicitly specified the `containerRuntime` in cluster config then it's a little hard, still do-able maybe, to determine if `containerd` was a result of defaulting or was explicitly set here.

Reply:
sounds good

Reply:
Sounds good.